File: 001.two-sum/two_sum.java
public class Solution {
  public int[] twoSum(int[] nums, int target) {
    HashMap<Integer, Integer> m = new HashMap<Integer, Integer>();
    int[] res = new int[2];
    for (int i = 0; i < nums.length; ++i) {
      m.put(nums[i], i);
    }
    for (int i = 0; i < nums.length; ++i) {
      int t = target - nums[i];
      if (m.containsKey(t) && m.get(t) != i) {
        res[0] = i;
        res[1] = m.get(t);
        break;
      }
    }
    return res;
  }
}

File: 001.two-sum/two_sum.cpp
class Solution {
public:
  vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> m;
    vector<int> res;
    for (int i = 0; i < nums.size(); ++i) {
      m[nums[i]] = i;
    }
    for (int i = 0; i < nums.size(); ++i) {
      int t = target - nums[i];
      if (m.count(t) && m[t] != i) {
        res.push_back(i);
        res.push_back(m[t]);
        break;
      }
    }
    return res;
  }
};
